core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go

```diff
@@ -0,0 +1,113 @@
+	emptyValue                           = []byte{}
+)
+
+type serviceInterpretationValueRepository struct {
+	enclaveDb          *enclave_db.EnclaveDB
+	starlarkValueSerde *kurtosis_types.StarlarkValueSerde
+}
+
+func getOrCreateNewServiceInterpretationTimeValueRepository(
+	enclaveDb *enclave_db.EnclaveDB,
+	starlarkValueSerde *kurtosis_types.StarlarkValueSerde,
+) (*serviceInterpretationValueRepository, error) {
+	if err := enclaveDb.Update(func(tx *bolt.Tx) error {
+		bucket, err := tx.CreateBucketIfNotExists(serviceInterpretationValueBucketName)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred while creating the bucket for the service interpretation time value repository")
+		}
+		logrus.Debugf("Recipe result bucket: '%+v'", bucket)
+
+		return nil
+	}); err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred while building service interpretation time value repository")
+	}
+
+	repository := &serviceInterpretationValueRepository{
+		enclaveDb:          enclaveDb,
+		starlarkValueSerde: starlarkValueSerde,
+	}
+
+	return repository, nil
+}
+
+func (repository *serviceInterpretationValueRepository) AddService(name service.ServiceName, service *kurtosis_types.Service) error {
+	logrus.Debugf("Saving service interpretation value '%v' for service with name '%v' to", service, name)
+	if err := repository.enclaveDb.Update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(serviceInterpretationValueBucketName)
+
+		serviceNameKey := getKey(name)
+		serializedValue := repository.starlarkValueSerde.Serialize(service)
+
+		// save it to disk
+		if err := bucket.Put(serviceNameKey, []byte(serializedValue)); err != nil {
+			return stacktrace.Propagate(err, "An error occurred while saving service interpretation time value '%v' for service '%v'", serializedValue, serviceNameKey)
+		}
+		return nil
+	}); err != nil {
+		return stacktrace.Propagate(err, "An error occurred while saving service interpretation time value '%v' for service '%v'", service, name)
+	}
+	logrus.Debugf("Succesfully saved service '%v'", name)
+	return nil
+}
+
+func (repository *serviceInterpretationValueRepository) GetService(name service.ServiceName) (*kurtosis_types.Service, error) {
+	logrus.Debugf("Getting service interpretation time value for service '%v'", name)
+	var value *kurtosis_types.Service
+
+	if err := repository.enclaveDb.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(serviceInterpretationValueBucketName)
+
+		serviceNameKey := getKey(name)
+
+		serviceSerializedValue := bucket.Get(serviceNameKey)
+
+		// check for existence
+		if serviceSerializedValue == nil {
+			return stacktrace.NewError("Service '%v' doesn't exist in the repository", name)
+		}
+
+		isEmptyValue := len(serviceSerializedValue) == len(emptyValue)
+
+		serviceSerializedValueStr := string(serviceSerializedValue)
+
+		// this will the case if the key was saved with an empty value
```

Comment:
`this will catch the case`?

Reply:
What cases would it be valid to store `service` with an empty value? If it's not valid, maybe we can prevent it from being stored as an empty value in `AddService`?

Reply:
Yeah, i had got this from recipe but now i am throwing an error!